zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+from datetime import datetime
+
+import flask
+from flask import request
+
+from .utils.json_result import json_result
+from .utils.route_wrappers import cross_domain, with_session
+from . import api
+import zeeguu
+from zeeguu.model import User, Cohort, UserActivityData, Session, Language
+import sqlalchemy
+from flask import jsonify
+import json
+import random
+import datetime
+
+# class function wrapper
+def class_function_checker(class_id):
+    from zeeguu.model import TeacherCohortMap
+    link = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+    for l in link:
+        if l.user_id == flask.g.user.id:
+            return True
+    return False
+
+
+# Takes user_id and returns user.name that corresponds
+#@api.route("/get_user_name/<id>", methods=["GET"])
+@with_session
+def get_user_name(id):
```

Comment:
this seems to be weak design.
seems to imply that if i'm logged in, 
and thus have a session, i can find out the name/id 
matching of everybody else in the system. 

i guess there should be a check whether the logged
in user is a teacher for the user for which we ask info 
about, right?

Reply:
also, seems to be superseded by /get_user_info below, right?

Reply:
Oh. I see. Cool! 
In that case maybe the underscore for private functions in the module would help!

Reply:
Nevermind, I had already changed get_user_info to make this function redundant. It is now removed!